src/rt/sched/status.h

```diff
@@ -0,0 +1,123 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "../ds/morebits.h"
+
+#include <ostream>
+
+namespace verona::rt
+{
+  /// TODO: backpressure docs
+  enum struct BackpressureState : uint8_t
+  {
+    Normal = 0b00,
+    Muted = 0b01,
+    Unmutable = 0b10,
+    MaybeUnmutable = 0b11,
+
+    IsUnmutable = 0b10,
+  };
+
+  inline bool operator&(BackpressureState s1, BackpressureState s2)
+  {
+    return (uint8_t)s1 & (uint8_t)s2;
+  }
+
+  inline std::ostream& operator<<(std::ostream& os, BackpressureState s)
+  {
+    switch (s)
+    {
+      case BackpressureState::Normal:
+        return os << "Normal";
+      case BackpressureState::Muted:
+        return os << "Muted";
+      case BackpressureState::Unmutable:
+        return os << "Unmutable";
+      case BackpressureState::MaybeUnmutable:
+        return os << "MaybeUnmutable";
+      default:
+        return os << "??";
+    }
+  }
+
+  /// Tracks status information for a cown. This class may only be modified by
+  /// the scheduler thread running a cown.
+  /// TODO: possibly I/O thread when cown is uncheduled
+  class Status
+  {
+    // Load at which cown is overloaded.
+    static constexpr uint32_t overload_threshold = 800;
```

Comment:
I imagine this could be dynamic depending on hardware specs, types of queues (ex. I/O vs. msg). This should probably be a multiplier `8:1` over a formula for the types, which could be hard-coded to `100` for now.